covalent/_shared_files/util_classes.py

```diff
@@ -19,34 +19,52 @@
 # Relief from the License may be granted by purchasing a commercial license.
 
 
-from dataclasses import dataclass
-from typing import NamedTuple
+import multiprocessing as mp
+import queue
+from multiprocessing.queues import Queue as MPQ
+from typing import Any, NamedTuple
 
+import cloudpickle as pickle
 
-@dataclass
-class Status:
-    STATUS: str
+from . import logger
 
-    def __bool__(self):
-        """
-        Return True if the status is not "NEW_OBJECT"
-        """
+app_log = logger.app_log
+log_stack_info = logger.log_stack_info
 
-        return self.STATUS != "NEW_OBJECT"
 
-    def __str__(self) -> str:
-        return self.STATUS
+class SafeVariable(MPQ):
+    def __init__(self) -> None:
+        super().__init__(maxsize=1, ctx=mp.get_context())
+
+    def save(self, value: Any) -> None:
+        value = pickle.dumps(value)
+        try:
+            self.put(value, timeout=0.5)
+        except queue.Full:
+            self.get_nowait()
+            self.put_nowait(value)
 
+    def retrieve(self) -> Any:
+        try:
+            value = self.get(timeout=0.5)
```

Comment:
Hi @kessler-frost, the `status_listener` would insert 0.5-second pauses in the event loop b/c `MPQ.get(timeout)` blocks the entire process.

Reply:
Yeah, that's true, my bad. I am now using a `queue.Queue` instead with `get_nowait()` and `put_nowait()` which should resolve this. Since `queue.Queue`'s are inherently threadsafe, this should better for our purpose (instead of using MPQ which also ensure process-safety).

Reply:
After our discussion, this has been now switched to an asyncio queue with added threadsafe saving and retrieving methods.